Resources/Shaders/colormatrix.swsl

```diff
@@ -0,0 +1,30 @@
+//This is for handling transformations
+//caused by DM colour matrices: https://secure.byond.com/docs/ref/#/{notes}/color-matrix
+//In general, our code should try to avoid using this shader, in favour of the 'modulate' Color parameter available in some RT/Clyde drawing routines.
+//This should only be used for color transformations that can only be expressed as the linear algebra nonsense featured below.
+
+uniform lowp mat4 colorMatrix;
+uniform lowp vec4 offsetVector; // The final row of the 4x5 matrix it kinda is
```

Comment:
```suggestion
uniform lowp vec4 offsetVector; // The final row of the 4x5 matrix it kinda is
uniform lowp int space; //color space flag - NOT IMPLEMENTED
```

Reply:
Please actually write TODO so futurecoders stumble upon it.

Reply:
I left `space` out here, in the shader (and put it in places in C# where it should be) because even the code design here is a bit of a TODO:
- I'm not confident that code that switches on that space arg is actually unbranching once compiled (and shaders with runtime branching are a no-no afaik). So switching should ideally be done by selecting which of a set of deterministic shaders to run (at which point `space` is not even a shader arg anymore)
- It's not always possible to convert (say) a HSV color matrix into an RGB one - many possible situations would be nonlinear. I have no clue what the correct solution to that is which is performant and does not involve a lot of slow trigonometry or something.

Reply:
You can runtime branch, see alpha filter for an example. I have no real idea how colour spaces work though, so I can't really respond to your second point. In truth I'd be amazed if anyone is actually using a non RGBA colour matrix for anything in our targets.